src/WtsApi32.Desktop/WtsApi32+WtsSafeMemoryGuard.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) to owners found in https://github.com/AArnott/pinvoke/blob/master/COPYRIGHT.md. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+
+namespace PInvoke
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Runtime.InteropServices;
+    using System.Text;
+
+    /// <content>
+    /// Exported functions from the WtsApi32.dll Windows library
+    /// that are available to Desktop apps only.
+    /// </content>
+    public static partial class WtsApi32
+    {
+        /// <summary>
+        /// Base guard for all memory allocated inside WtsApi32.
+        /// </summary>
+        public partial class WtsSafeMemoryGuard : SafeHandle
```

Comment:
Just enlighten me why does it descend from SafeHandle then? I think maybe there might be more appropriate structures and classes for this end. It is some memory that gets returned by a function or is a [Out] parameter pointer it most probably either is a full memory buffer or a handle to a memory buffer. On the first case you could use [`SafeBuffer`](https://msdn.microsoft.com/en-us/library/system.runtime.interopservices.safebuffer(v=vs.110).aspx) as the base class. This is something designed to safe memory buffers. Or we should let the user manage the thing. If it something you have to create/instantiate/allocate yourself and then pass it on as a parameter then it is not a `Safe` anything. Most, if not all, the `SafeHandle` specializations are returned by a function that initializes, starts or instantiates windows native objects and the way to identify it is through the Handle.